src/main/java/com/google/api/codegen/transformer/py/PythonPackageMetadataTransformer.java

```diff
@@ -69,23 +88,60 @@ public PythonPackageMetadataTransformer(PackageMetadataConfig packageConfig) {
         "py/setup.py.snip",
         "py/requirements.txt.snip",
         "py/README.rst.snip",
-        "py/tox.ini.snip");
+        "py/tox.ini.snip",
+        "py/docs/apis.rst.snip",
+        "py/docs/conf.py.snip",
+        "py/docs/index.rst.snip",
+        "py/docs/starting.rst.snip");
   }
 
   public List<String> getInitTemplateFileNames() {
     return Lists.newArrayList("py/__init__.py.snip", "py/namespace__init__.py.snip");
   }
 
   private ViewModel generateMetadataView(Model model, ApiConfig apiConfig, String template) {
-    String outputPath = Files.getNameWithoutExtension(template);
+    String noLeadingPyDir = template.startsWith("py/") ? template.substring(3) : template;
+    int extensionIndex = noLeadingPyDir.lastIndexOf(".");
+    String outputPath = noLeadingPyDir.substring(0, extensionIndex);
+    computeModules(gapicProviders);
 
     return metadataTransformer
         .generateMetadataView(packageConfig, model, template, outputPath, TargetLanguage.PYTHON)
-        .namespacePackages(
+        .pythonNamespacePackages(
             computeNamespacePackages(apiConfig.getPackageName(), packageConfig.apiVersion()))
+        .pythonApiModules(apiModules)
+        .pythonTypeModules(typeModules)
         .build();
   }
 
+  /** Determines the Python files generated in the main phase of generation. */
+  private void computeModules(List<GapicProvider<? extends Object>> gapicProviders) {
+    // Only run generation once.
+    if (apiModules != null && typeModules != null) {
+      return;
+    }
+    Set<String> apiModuleSet = new TreeSet<>();
```

Comment:
Done.

I think I was somehow concerned that different providers could produce duplicate output files, but if that ever happens, that's a larger issue to address...